### ISSUE TITLE
Accessibility Testing for [Check-in Experience, Check-in Experience, Travel Pay]

### ISSUE THREAD

**Issue by wullaski** [department-of-veterans-affairs/va.gov-team#58572]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@wullaski

### Use of color and color contrast checks

- [X] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [X] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [X] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [X] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [X] Any charts, maps, infographics, and tables convey all information without only relying on color
- [X] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### Who completed the axe scans?

@wullaski

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [X] axe is integrated into your end-to-end testing

### axe DevTools scan results

No issues found

### Who completed the content zoom and reflow test?

@wullaski

### Content zoom and reflow checks

- [X] All page elements are readable and usable at 200% zoom
- [ ] All page elements are readable and usable at 300% zoom
- [ ] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

There's an `overflow-x: hidden` applied to the body and content of the vets website that prevents horizontal scrolling. At zoom levels that are above 200% some words are too long to fit in a mobile sized viewport. Those words and elements get cut off and are unreadable.
![Screen Shot 2023-05-17 at 9 26 13 AM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/2982977/2fbdb4ca-0375-4496-be4c-c7a69d775bfc)

### Who completed the keyboard navigation test?

@wullaski

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [X] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### Do you have any other results to share?

Voiceover on mac went well. The only question I had was that if the va-alerts need to be called out or announced in some way.

**Comment by wullaski** [comment]
@briandeconinck I had a question about the va-alerts and screen readers. Should they be announced sort of like the loading messages?

**Comment by briandeconinck** [comment]
@wullaski Great question, and a complicated answer! 😄 

It really comes down to what the intention of the alert is, and whether the information needs to be presented to the user ASAP versus something that they can discover in the course of their normal user flow. So "something has gone wrong and requires action by you" may need to be announced immediately, but "hey you should be aware of this other thing" doesn't.

In general this corresponds with the alert types in the design system. Error alerts have [`role="alert"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role) coded into them out of the box, and anything with that role when the page loads will be the first thing that gets announced. Warning, information, and success alerts don't have `role="alert"` (which is admittedly confusing) and don't get that priority treatment by screen readers. But there are also exceptions where it's appropriate to add `role="alert"` to them because it would be confusing to users to not be prompted with some warning or success message.

If content is changing dynamically and it's useful to notify screen reader users of that change (alert or not), it may also be appropriate to use an [ARIA live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) --- and really, `role="alert"` is just one type of live region.

Let me know if that's useful, or if there's something specific you want to talk through.
